server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -456,22 +466,31 @@ private static void ensureToBeDecommissionedAttributeWeighedAway(ClusterState st
 
     private static void ensureEligibleRequest(
         DecommissionAttributeMetadata decommissionAttributeMetadata,
-        DecommissionAttribute requestedDecommissionAttribute
+        DecommissionRequest decommissionRequest
     ) {
-        String msg = null;
+        String msg;
+        DecommissionAttribute requestedDecommissionAttribute = decommissionRequest.getDecommissionAttribute();
         if (decommissionAttributeMetadata != null) {
             // check if the same attribute is registered and handle it accordingly
             if (decommissionAttributeMetadata.decommissionAttribute().equals(requestedDecommissionAttribute)) {
                 switch (decommissionAttributeMetadata.status()) {
-                    // for INIT and FAILED - we are good to process it again
+                    // for INIT - check if it is eligible internal retry
                     case INIT:
+                        if (decommissionRequest.requestID().equals(decommissionAttributeMetadata.requestID()) == false) {
+                            throw new DecommissioningFailedException(
+                                requestedDecommissionAttribute,
+                                "same request is already in status [INIT]"
```

Comment:
[minor]: another request instead of same request

Reply:
It's a same request from users perspective as it would be decommissioning same attribute name and attribute value. Although, technically its a different request but these messages are generally user friendly